core/keyboard_nav/cursor.js

```diff
@@ -94,6 +95,36 @@ Blockly.Cursor.prototype.hide = function() {
   }
 };
 
+/**
+ * Handles the given action.
+ * This is only triggered when keyboard navigation is enabled.
+ * @param {!Blockly.Action} action The action to be handled.
+ * @return {boolean} True if the action has been handled, false otherwise.
+ */
+Blockly.Cursor.prototype.onBlocklyAction = function(action) {
+  // If we are on a field give it the option to handle the action
+  if (this.curNode_ && this.curNode_.getType() === Blockly.ASTNode.types.FIELD &&
+      this.curNode_.getLocation().onBlocklyAction(action)) {
```

Comment:
user getCurNode() instead of accessing a private field directly since this is intended to be overridden.